src/sentry/migrations/0738_metric_alert_anomaly_detection.py

```diff
@@ -0,0 +1,93 @@
+# Generated by Django 5.0.6 on 2024-07-11 17:58
+
+from django.db import migrations, models
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.utils.translation import gettext_lazy
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapper
+
+
+class AlertRuleDetectionType(models.TextChoices):
+    STATIC = "Static", gettext_lazy("Static")
+    PERCENT = "Percent", gettext_lazy("Percent")
+    DYNAMIC = "Dynamic", gettext_lazy("Dynamic")
+
+
+def _backfill_detection_type(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    AlertRule = apps.get_model("sentry", "AlertRule")
+
+    # use RangeQuerySetWrapper to avoid loading all AlertRules into memory
+    for rule in RangeQuerySetWrapper(AlertRule.objects.all()):
```

Comment:
not sure what I am missing but when I try to run this locally it's failing saying `AttributeError: type object 'AlertRule' has no attribute 'objects'. Did you mean: 'projects'?` huh??

Reply:
The `AlertRule` model defines a custom default manager (`Meta.default_manager_name`), you need to use `AlertRule.objects_with_snapshots`